Sources/Client.swift

```diff
@@ -173,18 +183,66 @@ open class Client {
 
 
 extension Client {
+    /**
+     Fetch a collection of Entries from Contentful matching the specified query. This method does not
+     specify the content_type in the query parameters, so the entries returned in the results can be
+     of any type.
 
-    @discardableResult public func fetchEntries<ContentType: EntryModel>
-        (with query: Query<ContentType>, completion: @escaping (Result<[ContentType]>) -> Void) -> URLSessionDataTask? {
+     - Parameter query: The Query object to match results againts.
+     - Parameter completion: A handler being called on completion of the request.
 
-        let url = URL(forComponent: "entries", parameters: query.queryParameters())
+     - Returns: The data task being used, enables cancellation of requests.
+     */
+    @discardableResult public func fetchEntries(with query: Query, completion: @escaping ResultsHandler<[Entry]>) -> URLSessionDataTask? {
+
+        let url = URL(forComponent: "entries", parameters: query.parameters)
+        return fetch(url: url) { (result: Result<Array<Entry>>) in
+            switch result {
+            case .success(let array):
+
+                let entries = array.items
```

Comment:
Following the offline discussion we had: Do you think there is a need in exposing `total`, `skip`, `limit` of the result? Also I think right now yours is the only SDK returning just the items. `python` afaik, takes a python array and augemnts the requested fields on it...

(Similar Comment for the other api methods returning an Array)